### PR TITLE
Move generated scripts

### PR DESCRIPTION
**Pull request by ybznek**
I have done this separate PR due to amount of moved files.

After whole work you should be able to:
- generate file from shared template
- override the generated file using shared "static" file - e.g. have different remediation for a specific (meta)package.
- every product can override these shared "templates" by its own templates
- every product can override specific product template by static file

IOW: It should be like OOP inheritance, with 4 generations and optionally specialized behavior.

This is planned folder structure. 
```
Structure of templates

- shared/
 - templates/
   - csv/
   - oval_templates/
   - bash_templates/
 - static/
   - bash/
   - oval/
   - oval_5.11/

- PRODUCT/
 - templates/
   - csv/
   - oval_templates/
   - bash_templates/
 - static/
   - bash/
   - oval/
   - oval_5.11/
```
---
This file clone both repositories and tries to compare output datastreams.
It should REDUCE probability of change due to refactoring.
```
$ cat ./compare.sh 
#!/bin/bash

git clone https://github.com/OpenSCAP/scap-security-guide.git original && {
    pushd original 
    make -j4
    popd
}

git clone https://github.com/ybznek/scap-security-guide.git -b master-move update && {
    pushd update 
    make -j4
    popd
}

function removeDifferences() {
    sed 's;timestamp>[^<]*<;timepstamp><;g' | \
    sed 's;timestamp="[^"]*";;g' | \
    sed 's;date="[^"]*";"";g'

}

find ./original -name "*-ds.xml" | while read originalFile;
do
    toCompare=$(echo "$originalFile" | sed -E 's;^./original;./update;')
    echo "-----------------------------------------------------------------"
    echo "$originalFile <=> $toCompare" 
    echo "-----------------------------------------------------------------"
    
    cat "$originalFile" | removeDifferences | sort > /tmp/original
    cat "$toCompare"    | removeDifferences | sort > /tmp/new
    
    diff /tmp/original /tmp/new -q || {
        meld /tmp/original /tmp/new
        #exit
    }

done


```

### CODE REVIEW

**Comment by ybznek** [comment]
Files were moved from `remediations/bash -> templates/static/bash`

In next PRs should be files from static/bash "replaced" by generated scripts from templates

File: shared/product-make.include [hunk 1]
```diff
@@ -91,13 +91,13 @@ $(OUT)/xccdf-unlinked-ocilrefs.xml: $(OUT)/xccdf-unlinked-resolved.xml $(OUT)/oc
...
-bash_remediations_deps= $(wildcard $(IN)/remediations/bash/*.sh) $(wildcard $(SHARED_REMEDIATIONS)/*.sh)
+bash_remediations_deps= $(wildcard $(IN)/templates/static/bash/*.sh) $(wildcard $(SHARED_REMEDIATIONS)/*.sh)
 $(OUT)/bash-remediations.xml: $(SHARED)/$(TRANS)/combineremediations.py $(bash_remediations_deps)
 	# Make intermediate $(BUILD_REMEDIATIONS) directory to hold final list of remediation scripts for $(PROD)
 	mkdir -p $(BUILD_REMEDIATIONS)
-	# Search $(SHARED_REMEDIATIONS) and $(IN)/remediations directories to find all product specific remediation scripts,
+	# Search $(SHARED_REMEDIATIONS) and templates/static/bash directories to find all product specific remediation scripts,
 	# which are regular files (not symlinks). Merge the final list into $(BUILD_REMEDIATIONS) directory
-	find $(SHARED_REMEDIATIONS) $(IN)/remediations/bash -maxdepth 1 -type f -name '*.sh' -exec cp {} $(BUILD_REMEDIATIONS) ';'
+	find $(SHARED_REMEDIATIONS) templates/static/bash -maxdepth 1 -type f -name '*.sh' -exec cp {} $(BUILD_REMEDIATIONS) ';'
```
Comment (mpreisler) [inline shared/product-make.include]: why did you remove the `$(IN)` prefix here?

Comment (ybznek) [inline shared/product-make.include]: Because it was in input directory. I should be `product/templates/...` because shared dir uses format `shared/templates`. Do you think that it should be `product/input/templates`?

**Comment by iankko** [comment]
@ybznek 
ShellCheck is currently failing due to no files present in ```input/remediations/bash;``` directory on systems, where shellcheck check is performed:

```
cd input/remediations/bash; shellcheck -s bash *.sh
shellcheck: *.sh: openFile: does not exist (No such file or directory)
Makefile:73: recipe for target 'validate' failed
```
Could you either:
* Change the shellcheck test to look into new location, where the generated *.sh remediation scripts are intended to be found?, or
* Temporarily disable the shellcheck test, it not to return ```exit(1)``` till the repo rewrite is done (as the rewrite might cause the shellcheck won't work during some time).

Personally I like the former option better. But if there isn't other way, the latter is acceptable too for some time.

**Comment by mpreisler** [comment]
I strongly prefer to have this working through the refactoring. @ybznek please change the paths and keep the shellcheck running if at all possible.

**Comment by mpreisler** [comment]
ACK
